CondFormats/SiPixelTransient/src/SiPixelTemplate.cc

```diff
@@ -1950,6 +1909,207 @@ bool SiPixelTemplate::interpolate(int id, float cotalpha, float cotbeta, float l
+                               float locBz,
+                               float lorwdy,
+                               float lorwdx,
+                               float q50,
+                               float fbin[3],
+                               float xsize,
+                               float ysize,
+                               float zsize) {
+  // Set class variables to the input parameters
+  entry_sideloaded_ = entry;  // will bypass xtemp_[] and ytemp_[] and use those from this Entry.
+
+  enty1_ = entry;
+  enty0_ = entry;
+
+  entx00_ = entry;
+  entx02_ = entry;
+  entx20_ = entry;
+  entx22_ = entry;
+  entx21_ = entry;
+
+  Dtype_ = iDtype;
+  lorywidth_ = lorwdy;
+  lorxwidth_ = lorwdx;
+  xsize_ = xsize;
+  ysize_ = ysize;
+  zsize_ = zsize;
+  s50_ = q50;
+  qscale_ = 1.f;
+  for (int i = 0; i < 3; ++i) {
+    fbin_[i] = fbin[i];
+  }
+
+  // Set other class variables
+
+  // adcota_ = 0.f;
+  // adcotb_ = 0.f;
+
+  yratio_ = 0.f;
+  yxratio_ = 0.f;
+  xxratio_ = 0.f;
+
+  qavg_ = entry->qavg;
+  qmin_ = 0.f;
+  qmin2_ = 0.f;
+
+  pixmax_ = entry->pixmax;
+  sxmax_ = entry->sxmax;
+  symax_ = entry->symax;
+  clsleny_ = entry->clsleny;
+  clslenx_ = entry->clslenx;
+
+  scaleyavg_ = 1.f;
+  scalexavg_ = 1.f;
+  delyavg_ = 0.f;
+  delysig_ = 0.f;
+
+  dyone_ = entry->dyone;
+  syone_ = entry->syone;
+  dytwo_ = entry->dytwo;
+  sytwo_ = entry->sytwo;
+
+  dxone_ = entry->dxone;
+  sxone_ = entry->sxone;
+  dxtwo_ = entry->dxtwo;
+  sxtwo_ = entry->sxtwo;
+
+  chi2yminone_ = 0.f;
+  chi2yavgone_ = 0.1;
+  chi2xminone_ = 0.f;
+  chi2xavgone_ = 0.1;
+
+  for (int i = 0; i < 4; ++i) {
+    scalex_[i] = 1.f;
+    scaley_[i] = 1.f;
+    offsetx_[i] = 0.f;
+    offsety_[i] = 0.f;
+    xrms_[i] = 0.f;
+    yrms_[i] = 0.f;
+    xavg_[i] = 0.f;
+    yavg_[i] = 0.f;
+
+    chi2yavg_[i] = 0.1;
+    chi2xavg_[i] = 0.1;
+
+    chi2xmin_[i] = 0.f;
+    chi2ymin_[i] = 0.f;
+
+    for (int j = 0; j < 6; j++) {
+      yflparl_[i][j] = yflparh_[i][j] = entry->yflpar[i][j];
+      xflparhh_[i][j] = xflparhl_[i][j] = xflparll_[i][j] = xflparlh_[i][j] = entry->xflpar[i][j];
+    }
+  }
+
+  sxparmax_ = entry->sxmax;
+  syparmax_ = entry->symax;
+
+  // This works only for IP-related tracks
+
+  flip_x_ = false;
+  flip_y_ = false;
+  float cotbeta = entry->cotbeta;
+  switch (Dtype_) {
+    case 0:
+      if (cotbeta < 0.f) {
+        flip_y_ = true;
+      }
+      break;
+    case 1:
+      if (locBz > 0.f) {
+        flip_y_ = true;
+      }
+      break;
+    case 2:
+    case 3:
+    case 4:
+    case 5:
+      if (locBx * locBz < 0.f) {
+        flip_x_ = true;
+      }
+      if (locBx < 0.f) {
+        flip_y_ = true;
+      }
+      break;
+    default:
+      // #ifndef SI_PIXEL_TEMPLATE_STANDALONE
+      //         throw cms::Exception("DataCorrupt") << "SiPixelTemplate::illegal subdetector ID = " << Dtype_ << std::endl;
+      //#else
+      std::cout << "SiPixelTemplate:illegal subdetector ID = " << Dtype_ << std::endl;
```

Comment:
This is only used standalone, isn't it?

Reply:
Yes this is only used in standalone.  the commented out lines were leftovers, I removed them